great_tables/gt.py

```diff
@@ -314,6 +320,98 @@ def _render_formats(self, context: str) -> Self:
+                # The pattern syntax uses 1-based indexing so adjust here
+                col_idx = int(col_ref) - 1
+
+                # Check that the referenced column exists in the provided columns
+                if col_idx < 0:
+                    raise ValueError(
+                        f"Pattern references column {{{col_ref}}} but column indexing starts "
+                        f"at {{1}}, not {{0}}. Please use 1-based indexing in patterns."
+                    )
+                if col_idx >= len(columns):
+                    raise ValueError(
+                        f"Pattern references column {{{col_ref}}} but only {len(columns)} "
+                        f"columns were provided to cols_merge()."
+                    )
+
+            # Process each row (according to the `rows=` parameter in `cols_merge()`)
+            for row_idx in rows:
+                # Collect values and missing status for all columns
+                col_values = {}
+                col_is_missing = {}
+
+                for i, col_name in enumerate(columns):
+                    # Get the formatted value from the body
+                    formatted_value = _get_cell(new_body.body, row_idx, col_name)
+
+                    # Get the original value from the data table
+                    original_value = _get_cell(self._tbl_data, row_idx, col_name)
+
+                    # If the body cell is missing (unformatted) and the original has a value,
+                    # use the original value; otherwise use the formatted value
+                    if is_na(new_body.body, formatted_value) and not is_na(
+                        self._tbl_data, original_value
+                    ):
+                        # Cell is unformatted but has a value in the original data
+                        display_value = str(original_value)
+                    else:
+                        # If the cell is formatted OR the original is missing then use the
+                        # formatted value (which has the proper NA representation like "<NA>")
+                        display_value = str(formatted_value)
+
+                    # Store with 1-based index (as used in the pattern)
+                    col_key = str(i + 1)
+                    col_values[col_key] = display_value
+                    col_is_missing[col_key] = is_na(self._tbl_data, original_value)
+
+                # Process the pattern with the collected values
+                merged_value = _process_col_merge_pattern(
+                    pattern=pattern, col_values=col_values, col_is_missing=col_is_missing
+                )
+
+                # Set the merged value in the target column
+                result = _set_cell(new_body.body, row_idx, target_column, merged_value)
+
+                # For Pandas and Polars, _set_cell() modifies in place and returns None but
+                # for PyArrow, _set_cell() returns a new table
+                if result is not None:
```

Comment:
Does this custom handling of backends happen in other places outside of `_tbl_data.py`?